### PR TITLE
Null ability updated

### PR DESCRIPTION
**Pull request by RonAzar**

### CODE REVIEW

File: src/main/kotlin/NullAbilityUpdated/Keren/Q4.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        Employee("Jane", null),
+        Employee(null, Address(null, "Unknown Street")),
+        Employee("Alice", Address("Los Angeles", null))
+    )
+
+    // Task: Print each employee's city safely. If the city is not available, print "City Unknown".
+    employees.printsAllEmployeesCites()
+}
+
+fun List<Employee>.printsAllEmployeesCites() = this.forEach( { println(it.address?.city ?: "City Unknown") })
```
Comment (kerensegev) [inline src/main/kotlin/NullAbilityUpdated/Keren/Q4.kt:24]: This shouldn't be an extension function on  List<Employee>, can be a regular function

File: src/main/kotlin/NullAbilityUpdated/Keren/Q5.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        Product(null, 299.99),
+        Product("Smartphone", null),
+        Product(null, null)
+    )
+
+    // Task: Print the details of products only if both name and price are not null.
+    products.printsAllProductsDetails()
+}
+
+fun List<Product?>.printsAllProductsDetails() {
```
Comment (kerensegev) [inline src/main/kotlin/NullAbilityUpdated/Keren/Q5.kt:25]: shouldn't be an extension function

File: src/main/kotlin/NullAbilityUpdated/Keren/Q6.kt [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        Department(null, EmployeeDetails("Bob", ContactInfo(null, "123-456-7890"))),
+        Department("Marketing", EmployeeDetails(null, ContactInfo("marketing@example.com", "987-654-3210")))
+    )
+
+    // Task: Print each department's name and manager's contact information.
+    // If any information is missing, use appropriate defaults.
+    departments.printEachDepartmentNameAndContactInfo()
+}
+
+fun List<Department?>.printEachDepartmentNameAndContactInfo() {
```
Comment (kerensegev) [inline src/main/kotlin/NullAbilityUpdated/Keren/Q6.kt:20]: here also

**Review by kerensegev — APPROVED** [review]

File: src/main/kotlin/NullAbilityUpdated/Keren/Q4.kt (hunk 1 above)
Comment (RonAzar) [reply src/main/kotlin/NullAbilityUpdated/Keren/Q4.kt:24]: Got it
Yes, Keren, I know it shouldn’t be an extension—I was just practicing. 
Is there anything else you think could be improved aside from the unnecessary extension? 
Thanks again for the review!

**Pull request merged**